### PR TITLE
Create proper systemd::unit for critical_units target

### PR DESCRIPTION
**Pull request by tomaszdurka**

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@kris-lab or @njam please review

File: modules/systemd/manifests/unit.pp [hunk 1]
```diff
@@ -1,11 +1,15 @@
 define systemd::unit(
   $content,
+  $type = 'service',
+  $critical = true,
 ) {
+  
+  $unitName = "${name}.${type}"
```
Comment (kris-lab) [inline modules/systemd/manifests/unit.pp]: maybe better `unit_name`?

**Comment by kris-lab** [comment]
What this new `critical` param exactly does? Is it to avoid new nested unit?

**Comment by tomaszdurka** [comment]
Yes otherwise there was critical-units wants entry for critical-units. On the other hand it didn't make difference... but I would not create that file.

**Comment by tomaszdurka** [comment]
@kris-lab I take that as ok! :)

please confirm!

File: modules/systemd/manifests/unit.pp (hunk 1 above)
Comment (tomaszdurka) [inline modules/systemd/manifests/unit.pp]: Am confused by our conventions. Could and can see some camelcases, but also undercored ones. I see that puppet core has underscores so let's use it.

**Comment by kris-lab** [comment]
yep, that's fine!

**Pull request merged**